### PR TITLE
Fixes #19089: Add a rudder webapp system technique

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/19089

### CODE REVIEW

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
Missing license in new .cf files

**Comment by Fdall** [comment]
PR updated with a new commit

File: techniques/system/rudderWebapp/1.0/integrity-check.cf [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+bundle agent system_root_integrity_check(trackingkey, technique_name)
```
Comment (ncharles) [inline techniques/system/rudderWebapp/1.0/integrity-check.cf:18]: you could fetch the values from bundle system_rudder_webapp, wouldn't it be better ?

File: techniques/system/rudderWebapp/1.0/integrity-check.cf [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      "pass2" expression => "pass1";
+      "pass1" expression => "any";
+
+  files:
+      "${g.rudder_configuration_repository}"
+        create => "true",
+        action => WarnOnly,
+        classes => if_else("rudder_integrity_ok", "rudder_integrity_failed");
+
+      "${g.rudder_configuration_repository}/.git/index.lock"
```
Comment (ncharles) [inline techniques/system/rudderWebapp/1.0/integrity-check.cf:38]: What happens if we have a slow git command running ?

**Review by ncharles — COMMENTED** [review]
Looks good to me - i'm simply unsure about passing the technique name and the tracking key

File: techniques/system/rudderWebapp/1.0/integrity-check.cf (hunk 2 above)
Comment (Fdall) [reply techniques/system/rudderWebapp/1.0/integrity-check.cf:38]: I believe that we should avoid when possible to rely on data taken from other resources.

File: techniques/system/rudderWebapp/1.0/integrity-check.cf (hunk 1 above)
Comment (Fdall) [reply techniques/system/rudderWebapp/1.0/integrity-check.cf:18]: I believe that we should avoid when possible to rely on data taken from other resources.

File: techniques/system/rudderWebapp/1.0/integrity-check.cf (hunk 2 above)
Comment (Fdall) [reply techniques/system/rudderWebapp/1.0/integrity-check.cf:38]: Wrong comment, I have no idea, I just took the existing techniques and port them.

**Comment by Fdall** [comment]
PR updated with a new commit

**Review by ncharles — APPROVED** [review]

**Comment by Fdall** [comment]
OK, squash merging this PR

**Pull request merged**